app/src/main/java/unipd/se18/ocrcamera/CameraActivity.java

```diff
@@ -0,0 +1,446 @@
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_camera);
+        // Create the directories for the app
+        createOCRCDirs();
+        // Initializing of the UI components
+        mCameraTextureView = findViewById(R.id.camera_view);
+        Button mButtonTakePhoto = findViewById(R.id.take_photo_button);
+        mButtonTakePhoto.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                takePhoto();
+            }
+        });
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        Log.e(TAG, "onResume");
+        //startBackgroundThread();
+        if (mCameraTextureView.isAvailable()) {
+            openCamera();
+        } else {
+            mCameraTextureView.setSurfaceTextureListener(mSurfaceTextureListener);
+        }
+    }
+
+    @Override
+    protected void onPause() {
+        closeCamera();
+        //stopBackgroundThread();
+        super.onPause();
+    }
+
+    /**
+     * Opens a connection with a camera if it is permitted, otherwise return.
+     */
+    private void openCamera() {
+        CameraManager manager = (CameraManager) getSystemService(Context.CAMERA_SERVICE);
+        Log.e(TAG, "is camera open");
+        try {
+            mCameraId = manager.getCameraIdList()[0];
+            CameraCharacteristics characteristics = manager.getCameraCharacteristics(mCameraId);
+            StreamConfigurationMap map = characteristics.get(CameraCharacteristics.SCALER_STREAM_CONFIGURATION_MAP);
+            assert map != null;
+            mPreviewSize = map.getOutputSizes(SurfaceTexture.class)[0];
+            // Add permission for camera and let user grant the permission
+            if (ActivityCompat.checkSelfPermission(this, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+                ActivityCompat.requestPermissions(CameraActivity.this, new String[]{Manifest.permission.CAMERA, Manifest.permission.WRITE_EXTERNAL_STORAGE}, MY_CAMERA_REQUEST_CODE);
+                return;
+            }
+            manager.openCamera(mCameraId, mStateCallback, null);
+        } catch (CameraAccessException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Controls the output of the permissions requests.
+     * @param requestCode The code assigned to the request
+     * @param permissions The list of the permissions requested
+     * @param grantResults The results of the requests
+     */
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String permissions[], @NonNull int[] grantResults) {
+        if (requestCode == MY_CAMERA_REQUEST_CODE) {
+            if (grantResults[0] == PackageManager.PERMISSION_DENIED) {
+                // close the app
```

Comment:
Thinking to user's experience it's not good that if permissions aren't granted the app closes. 
Why don't you only show a message to the user to alert him that without permissions the app won't work?